src/analyze/analyze-verify.c

```diff
@@ -233,6 +248,16 @@ int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run
         if (strv_isempty(filenames))
                 return 0;
 
+        /* Allow systemd-analyze to hook in a callback function so that it can get
+         * all the required log data from the function itself without having to rely
+         * on a global set variable for the same */
+        if (return_error_on) {
```

Comment:
We generally don't use boolean comparisons for non-booleans, see CODING_STYLE. `return_error_on != RETURN_ERROR_ON_NONE` ?